src/cli/cmd_clean.py

```diff
@@ -0,0 +1,150 @@
+    # Get unverified files by the index of the verification results list.
+    unverified_dfs = [
+        datafiles[ind] for ind, is_verified in enumerate(results) if not is_verified
+    ]
+
+    if len(unverified_dfs) > 0:
+        logger.info("Datafiles pending ingestion or verification:")
+        for datafile in unverified_dfs:
+            logger.info(datafile.directory / datafile.filename)
+        logger.error(
+            "Could not proceed with deleting this data root. Ingestion is not complete."
+        )
+        sys.exit(1)
+    else:
+        logger.info("Ingestion for this data root is complete.")
+
+
+def _check_file_age(df_paths: list[Path], min_file_age: int) -> None:
+    logger.info("Checking file age...")
+    newest_ctime = max_ctime(df_paths)
+    days_since_ctime = (datetime.now() - datetime.fromtimestamp(newest_ctime)).days
+    if not days_since_ctime >= min_file_age:
+        logger.error(
+            "Could not proceed with deleting this data root. Files were created %i days ago"
+            + ", but min_file_age is %i days.",
+            days_since_ctime,
+            min_file_age,
+        )
+        sys.exit(1)
+    else:
+        logger.info(
+            "File age is %i days, which meets min_file_age of %i days.",
+            days_since_ctime,
+            min_file_age,
+        )
+
+
+def _delete_datafiles(df_paths: list[Path]) -> None:
+    logger.info("Deleting datafiles.")
+    for pth in df_paths:
+        if pth.exists() and pth.is_file():
+            logger.info("Deleting %s", pth)
+            try:
+                pth.unlink()
+            except OSError:
+                logger.exception("File could not be deleted: %s", pth)
+                logger.error("Deleting verified files could not finish, exiting.")
+                sys.exit(1)
+        else:
+            logger.warning(
+                "File does not exist or isn't a file, skipping: %s",
+                pth,
+            )
+
+
+def clean(
+    source_data_path: SourceDataPathArg,
+    profile_name: ProfileNameOption,
+    storage: StorageBoxOption = None,
+    profile_version: ProfileVersionOption = None,
+    ask_first: Annotated[
+        Optional[bool],
+        typer.Option(
+            help=(
+                "Ask whether the datafiles should be deleted, before attempting"
+                " to delete the whole data root. Defaults to yes."
+            ),
+        ),
+    ] = True,
+    min_file_age: Annotated[
+        Optional[int],
+        typer.Option(
+            help=(
+                "Minimum age of files, in days, before we try to delete the data"
+                " root. Defaults to no minimum age."
+            )
+        ),
+    ] = None,
+    log_file: LogFileOption = Path("clean.log"),
+) -> None:
+    """Delete datafiles in source data root after ingestion is complete."""
+    log_utils.init_logging(file_name=str(log_file), level=logging.DEBUG)
```

Comment:
Do we want to hardcode the logging level?

Reply:
Removed log level hardcoding.